Sources/SwiftGenKit/Parsers/Filter.swift

```diff
@@ -33,6 +33,17 @@ public struct Filter {
   }
 }
 
+extension Parser {
+  /// Create the regex for a filter for the given list of file extensions. The extensions will be matched at the end
+  /// of a filename, and will be matched case-insensitively.
+  ///
+  /// - Parameter extensions: The list of file extensions.
+  /// - Returns: The regex as a `String`.
+  static func filterRegex(forExtensions extensions: [String]) -> String {
+    "[^/]\\.(?i:\(extensions.map { NSRegularExpression.escapedPattern(for: $0) }.joined(separator: "|")))$"
```

Comment:
nit for regex readability:
```suggestion
    let extensionsList = extensions.map { NSRegularExpression.escapedPattern(for: $0) }.joined(separator: "|")
    // - require at list one character that is not a "/" before the "." in order to avoid matching dotfiles
    // - then a literal "."
    // - then one of the accepted list of extensions, in a case-insensitive way (`(?:i…)`), at the end of the string (`$`)
    return "[^/]\\.(?i:\(extensionsList))$"
```